connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectStubState.scala

```diff
@@ -28,15 +28,15 @@ import org.apache.spark.internal.Logging
 // logic.
 class SparkConnectStubState(
     channel: ManagedChannel,
-    val retryPolicy: GrpcRetryHandler.RetryPolicy)
+    val retryHandler: GrpcRetryHandler)
     extends Logging {
 
+  def this(channel: ManagedChannel, retryPolicies: Seq[GrpcRetryHandler.RetryPolicy]) =
+    this(channel, new GrpcRetryHandler(retryPolicies))
+
   // Responsible to convert the GRPC Status exceptions into Spark exceptions.
   lazy val exceptionConverter: GrpcExceptionConverter = new GrpcExceptionConverter(channel)
 
-  // Manages the retry handler logic used by the stubs.
-  lazy val retryHandler = new GrpcRetryHandler(retryPolicy)
-
```

Comment:
I think I like it more like that, one retryHandler shared across everyone who wants it (why would you reinstantiate again)? If it would be like that, this Pr wouldn't have even need to touch this class.

Reply:
Reverting these changes, there is still one retryHandler shared across everyone who wants it.
There is only one SparkConnectStubState per client (that's why it got refactored out of BlockingStub, and Stub - so that all the stubs etc. and every use of it within a client share it). So it's only one retryHandler either way.
If you have multiple clients, they have multiple retryHandlers anyway.

The only place (outside testing) that creates it is in SparkConnectClient, uses this constructor anyway:
```
private[this] val stubState = new SparkConnectStubState(channel, configuration.retryPolicy)
```

so the only change that changes in this files do is:
* make retryHandler a `val` instead of `lazy val` - since the other objects in this state are lazy, it would be more consistent if it stayed lazy.
* have two constructors, and the new default one passing the GrpcRetryHandler is unused - why have an unused constructor?